File: projects/changestatus-bc/src/main/java/com/robinfood/changestatusbc/mappers/OrderBrandHistoryMappers.java
package com.robinfood.changestatusbc.mappers;

import com.robinfood.changestatusbc.dtos.OrderBrandHistoryDTO;
import com.robinfood.changestatusbc.entities.OrderBrandHistoryEntity;

public final class OrderBrandHistoryMappers {

    private OrderBrandHistoryMappers() {
        throw new IllegalStateException("Utility class");
    }

    public static OrderBrandHistoryEntity toOrderBrandHistoryEntity(OrderBrandHistoryDTO orderBrandHistoryDTO) {
        return OrderBrandHistoryEntity.builder()
                .brandId(orderBrandHistoryDTO.getBrandId())
                .orderId(orderBrandHistoryDTO.getOrderId())
                .orderStatusId(orderBrandHistoryDTO.getOrderStatusId())
                .userId(orderBrandHistoryDTO.getUserId())
                .build();
    }
}

File: projects/order-bc/app/src/main/java/com/robinfood/app/mappers/input/OrderBrandHistoryMappers.kt
@file:JvmName("OrderBrandHistoryMappers")

package com.robinfood.app.mappers.input

import com.robinfood.core.dtos.request.order.OrderBrandHistoryDTO
import com.robinfood.core.entities.OrderBrandHistoryEntity

fun OrderBrandHistoryDTO.toOrderBrandHistoryEntity(): OrderBrandHistoryEntity{
    return OrderBrandHistoryEntity(
        brandId,
        null,
        null,
        orderId,
        orderStatusId,
        null,
        userId
    );
}
